src/NodaTime/XmlSchemaDefinition.cs

```diff
@@ -0,0 +1,241 @@
+            IntervalSchemaType = new Lazy<XmlSchemaType>(() => new XmlSchemaComplexType
+            {
+                Name = nameof(Interval),
+                Attributes = {
+                    new XmlSchemaAttribute { Name = "start", SchemaTypeName = InstantSchemaType.Value.QualifiedName },
+                    new XmlSchemaAttribute { Name = "end", SchemaTypeName = InstantSchemaType.Value.QualifiedName }
+                }
+            });
+            LocalDateSchemaType = new Lazy<XmlSchemaType>(() => CreateSchemaTypeWithAttributes<LocalDate>(localDateRestriction.Value, calendarAttribute.Value));
+            LocalDateTimeSchemaType = new Lazy<XmlSchemaType>(() => CreateSchemaTypeWithAttributes<LocalDateTime>(localDateTimeRestriction.Value, calendarAttribute.Value));
+            LocalTimeSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<LocalTime>(xsStringType, TimePattern));
+            OffsetSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<Offset>(xsStringType, OffsetPattern));
+            OffsetDateSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<OffsetDate>(xsStringType, $"{YearPattern}-{MonthPattern}-{DayPattern}{OffsetPattern}"));
+            OffsetDateTimeSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<OffsetDateTime>(xsStringType, $"{YearPattern}-{MonthPattern}-{DayPattern}T{TimePattern}{OffsetPattern}"));
+            OffsetTimeSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<OffsetTime>(xsStringType, $"{TimePattern}{OffsetPattern}"));
+            PeriodBuilderSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<PeriodBuilder>(xsStringType, PeriodBuilderPattern));
+            YearMonthSchemaType = new Lazy<XmlSchemaType>(() => CreatePatternRestriction<YearMonth>(xsStringType, $"{YearPattern}-{MonthPattern}"));
+            ZonedDateTimeSchemaType = new Lazy<XmlSchemaType>(() => CreateSchemaTypeWithAttributes<ZonedDateTime>(OffsetDateTimeSchemaType.Value, zoneAttribute.Value, calendarAttribute.Value));
+
+            DependentSchemaTypes = new Lazy<Dictionary<XmlSchemaType, IEnumerable<XmlSchemaType>>>(() =>
+                new Dictionary<XmlSchemaType, IEnumerable<XmlSchemaType>>
+                {
+                    [IntervalSchemaType.Value] = new[] { InstantSchemaType.Value },
+                    [LocalDateSchemaType.Value] = new[] { localDateRestriction.Value, calendarRestriction.Value },
+                    [LocalDateTimeSchemaType.Value] = new[] { localDateTimeRestriction.Value, calendarRestriction.Value },
+                    [ZonedDateTimeSchemaType.Value] = new[] { OffsetDateTimeSchemaType.Value, calendarRestriction.Value, zoneRestriction.Value },
+                }
+            );
+
+            XmlSchema = new Lazy<XmlSchema>(CreateNodaTimeXmlSchema);
+        }
+
+        private static readonly Lazy<XmlSchema> XmlSchema;
```

Comment:
Do these definitely need to be lazy? If we're only going to get here when XML schemas are being used (which I suspect is a small subset of code) is it worth doing?

Reply:
Not sure why I went crazy with all these lazy. I removed all the lazy types and created everything in the static constructor. The resulting code is much less cluttered!